client/main.js

```diff
@@ -233,6 +258,116 @@ class ReputationMiningClient {
     return key;
   }
 
+  /**
+   * For update `_i` in the reputationUpdateLog currently under consideration, return the log entry that contains that update. Note that these
+   * are not the same number because each entry in the log implies multiple reputation updates.
+   * @param  {Number}  _i The update number we wish to determine which log entry in the reputationUpdateLog creates
+   * @return {Promise}   A promise that resolves to the number of the corresponding log entry.
+   */
+  async getLogEntryNumberForUpdateNumber(_i) {
+    const updateNumber = new BN(_i.toString());
+    const addr = await this.colonyNetwork.getReputationMiningCycle(true);
+    const repCycle = new ethers.Contract(addr, ReputationMiningCycleJSON.abi, this.realWallet);
+    const nLogEntries = await repCycle.getReputationUpdateLogLength();
+    let lower = new BN("0");
+    let upper = new BN(nLogEntries.toString()).subn(1);
+
+    while (!upper.eq(lower)) {
+      const testIdx = lower.add(upper.sub(lower).divn(2));
```

Comment:
We do have rounding error here, and it makes this less efficient than it would otherwise be, but we do get past log entry 1.

`lower=0; upper=2`.
First iteration:
`testIdx = 0 + ((2 - 0)/2) => 1`
`lower +=1` (for the sake of argument)
Second iteration:
`testIdx = 1 + ((2 - 1)/2) => 1`
`lower +=1` (same as above, because it's the same `testIdx`)
Third iteration:
`testIdx = 2 + ((2 - 2)/2) => 2`

Presumably the loop finds what it's looking for here, otherwise we're in real trouble (as this is the last remaining log entry).

Reply:
Actually went and looked at this code (which now lives [here](https://github.com/JoinColony/colonyNetwork/pull/212/files#diff-9c228e1a1f7fe6683164300c770b62ffR300), and you've misunderstood what happens. `lower` and `upper` aren't incremented/decremented, they're set to `testIdx+1` or `testIdx-1`. So in your examples, `lower` would become `2` and `upper` would become `0`